### PR TITLE
Uppy Url Plugin

### PR DESCRIPTION
**Pull request by SanderGi**
Task: https://app.asana.com/0/1203067047205953/1206283711518041
Server PR: https://github.com/GooeyAI/gooey-server/pull/261

### CODE REVIEW

File: app/gooeyFileInput.tsx [hunk 1]
```diff
@@ -12,6 +12,260 @@ import dashboardStyle from "@uppy/dashboard/dist/style.min.css";
...
+  }
+  urlItems.forEach(item => {
+    item.getAsString((urlString: any) => callback(urlString));
+  });
+}
+
+function _classPrivateFieldLooseBase(receiver: any, privateKey: any) { if (!Object.prototype.hasOwnProperty.call(receiver, privateKey)) { throw new TypeError("attempted to use private field on non-instance"); } return receiver; }
+var id = 0;
+function _classPrivateFieldLooseKey(name: any) { return "__private_" + id++ + "_" + name; }
+var _handleSubmit = /*#__PURE__*/_classPrivateFieldLooseKey("handleSubmit");
```
Comment (devxpy) [inline app/gooeyFileInput.tsx]: Is this from a compiler or something

Comment (SanderGi) [reply app/gooeyFileInput.tsx]: It is copypasted from the existing uppy url plugin, however they hardcode having to pay them to upload files via transloadit, so it was easier to just make a few modifications to their source code to replace that dependency. It’s MIT licensed so this should be allowed. They probably used some sort of transpiler

Comment (SanderGi) [reply app/gooeyFileInput.tsx]: I have removed all weird code like this and added proper types to everything. Also removed as much hacky code as possible and added comments for anything nontrivial. Lmk if there's anything else I can do to improve this pr or help get it production ready.

**Comment by SanderGi** [comment]
From Dara:
image

Button (Clear All) issue has been fixed: 
image

https://github.com/GooeyAI/gooey-ui/assets/97496861/48c95aa4-a15f-49ad-b01e-90dff022d81c

**Comment by SanderGi** [comment]
Closing since something functional has been deployed

**Pull request closed without merging**